### ISSUE TITLE
Akka.Cluster.Tools.Singleton.ClusterSingletonManager bug: An element with the same key but a different value already exists

### ISSUE THREAD

**Issue by denisTarasovRealine** [akkadotnet/akka.net#4474]
- Which Akka.Net version you are using

"Akka" Version="1.4.7" 
"Akka.Cluster" Version="1.4.7"
"Akka.Cluster.Tools" Version="1.4.7"

- On which platform you are using Akka.Net

Kubernates, Debian GNU/Linux 10

- A list of steps to reproduce the issue. Or an gist or github repo which can be easily used to reproduce your case.

Akka cluster deployed in kubernetes with lighthouse as seed nodes. Type of deployment is StatefulSet. Have a periodicaly issues like this: 

"ClassName": "System.ArgumentException",
"Message": "An element with the same key but a different value already exists. Key: UniqueAddress: (akka.tcp://calccluster@calc-batching-0.calc-batching:5213, 1739866769)",
"Source": "System.Collections.Immutable",
"StackTraceString": 
"at System.Collections.Immutable.ImmutableDictionary`2.HashBucket.Add(TKey key, TValue value, IEqualityComparer`1 keyOnlyComparer,IEqualityComparer`1 valueComparer, KeyCollisionBehavior behavior, OperationResult& result)
at System.Collections.Immutable.ImmutableDictionary`2.Add(TKey key, TValue value, KeyCollisionBehavior behavior, MutationInput origin) 
at System.Collections.Immutable.ImmutableDictionary`2.Add(TKey key, TValue value) 
at Akka.Cluster.Tools.Singleton.ClusterSingletonManager.AddRemoved(UniqueAddress node)
at Akka.Cluster.Tools.Singleton.ClusterSingletonManager.<InitializeFSM>b__42_1(Event`1 e)
at Akka.Actor.FSM`2.ProcessEvent(Event`1 fsmEvent, Object source)
at Akka.Actor.FSM`2.ProcessMsg(Object any, Object source) 
at Akka.Actor.FSM`2.Receive(Object message)
at Akka.Actor.ActorBase.AroundReceive(Receive receive, Object message)
at Akka.Actor.ActorCell.ReceiveMessage(Object message)
at Akka.Actor.ActorCell.Invoke(Envelope envelope)"

**Comment by Aaronontheweb** [comment]
Thanks @denisTarasovRealine - looks like a programming error with the `ClusterSingletonManager`

**Comment by Arkatufus** [comment]
@denisTarasovRealine,
Can you provide a simplified project to reproduce the bug?

**Comment by Aaronontheweb** [comment]
@Arkatufus 

> Akka.Cluster.Tools.Singleton.ClusterSingletonManager.AddRemoved(UniqueAddress node) 

That looks like the line that threw - we should look at it and see where we can have an idempotency error there.

**Comment by denisTarasovRealine** [comment]
I installed a role for a singleton, but forgot to set a role for a singleton manager. That caused the bug. After I fixed it, the error disappeared

 ```sh
ActorSystem systemLocal = x.Resolve<ActorSystem>();
systemLocal.ActorOf(ClusterSingletonProxy.Props(
    singletonManagerPath: $"/user/{LoadBalancer.Name}",
    settings: ClusterSingletonProxySettings
        .Create(systemLocal)
        .WithRole(AkkaConfiguration.Role)
        .WithSingletonIdentificationInterval(TimeSpan.FromSeconds(1))
        .WithBufferSize(8192)), 
        name: $"{LoadBalancer.Name}Proxy");
        
systemLocal.ActorOf(ClusterSingletonManager.Props(
        resolver.Create<LoadBalancer>(),
        PoisonPill.Instance,
        ClusterSingletonManagerSettings.Create(systemLocal)
            .WithRole(AkkaConfiguration.Role)), // I added this line and it worked
            name: LoadBalancer.Name);
```

**Comment by Aaronontheweb** [comment]
closed via #4489
